lib/node/node.js

```diff
@@ -471,9 +478,8 @@ function validate(validator, params, expectedParamName) {
 }
 
 Node.prototype.checkLimits = function(context, callback) {
-    // the default is to limit number of output rows if a maximumNumberOfRows is
-    // present in the configuration (not by default)
-    var defaultLimit = null;
-    var limit = context.getLimit(this.getType(), 'maximumNumberOfRows', defaultLimit, 'too many result rows');
-    limits.limitNumberOfRows(this, context, limit, callback);
+    Checks.check(this, context, [{
```

Comment:
Having this here allow us to limit the number of result rows for any analysis type by setting a limit value in the configuration.

If there's no configuration value, since where have `defaultLimit` here the behaviour is equivalent to `callback(null)` (because `limitNumberOfRows` does exactly that for null/Infinite limit values).